composer/datasets/streaming/download.py

```diff
@@ -80,72 +61,32 @@ def download(remote: str, local: str, timeout: float) -> None:
     """
     local_dir = os.path.dirname(local)
     os.makedirs(local_dir, exist_ok=True)
-    start_time = time()
 
     if remote.startswith('s3://'):
-        download_from_s3(remote, local, timeout=timeout)
+        download_from_s3(remote, local, timeout)
     else:
         download_from_local(remote, local)
 
-    elapsed = time() - start_time
-    if elapsed > timeout:
-        raise TimeoutError(f'Waited too long (more than {timeout:.3f} sec) for download')
 
-
-def safe_download(remote: str, local: str, timeout: float = 60) -> None:
+def safe_download(remote: str, local: str, timeout: float) -> None:
     """Safely downloads a file from remote to local.
        Handles multiple threads attempting to download the same shard.
        Gracefully deletes stale tmp files from crashed runs.
 
-
     Args:
         remote (str): Remote path (S3 or local filesystem).
         local (str): Local path (local filesystem).
-        timeout (float): How long to wait for shard to download before raising an exception. Default: 60 sec.
+        timeout (float): How long to wait for shard to download before raising an exception.
     """
-    # If we already have the file cached locally, we are done.
     if os.path.exists(local):
         return
 
-    # Check if there is a tmp file.
-    local_tmp = local + '.tmp'
-    if os.path.exists(local_tmp):
-        # Get tmp file created time
-        local_tmp_create_time = os.path.getctime(local_tmp)
-
-        # Get current disk time, more consistent than system time
-        with tempfile.NamedTemporaryFile() as f:
-            current_disk_time = os.path.getctime(f.name)
-
-        if current_disk_time - local_tmp_create_time < timeout + 1:  # 1s buffer to avoid race condition
-            # If the tmp file is recent, it is either (1) from a very recent crashed run, or (2) another thread is actively downloading it.
-            # So we wait but don't error out, in case we are in situation (1)
-            try:
-                wait_for_download(local, timeout)
-                return
-            except TimeoutError:
-                pass
-
-        # The tmp file is old, it is either (1) from a crashed run or (2) another thread is downloading it but is taking too long, and will timeout.
-        # Let's delete the tmp file. If situation (1), this is safe. If situation (2), the other thread is expected to crash with a TimeoutError anyways, so this is fine.
+    ok = False
+    for _ in range(3):
```

Comment:
Could we make this `3` a `max_retries` parameter and potentially lift it up to the `StreamingDataset`?

Reply:
Done